tokio/src/io/read_buf.rs

```diff
@@ -39,7 +39,7 @@ impl<'a> ReadBuf<'a> {
         }
     }
 
-    /// Creates a new `ReadBuf` from a fully uninitialized buffer.
+    /// Creates a new `ReadBuf` from a buffer that may be fully uninitialized.
     ///
     /// Use `assume_init` if part of the buffer is known to be already initialized.
```

Comment:
I think it would be nice to reword this to something along these lines:

1. The entire buffer will be marked as uninitialized by the internal cursor.
2. If the buffer is known to be partially initialized, then you can use `assume_init` to move the internal cursor.

or something like that.